### DISCUSSION TITLE
Code formatting

### DISCUSSION THREAD

**Discussion by quantumm** [PyMoDAQ/PyMoDAQ#43] [General]
Hi,

I would like to draw attention on code formatting. I think the code of PyMoDAQ would gain a lot of readability by using a common code formatting.

I discovered recently a tool called Black that do the job automatically ! [https://dev.to/mblayman/consistent-python-code-with-black-4c87](https://dev.to/mblayman/consistent-python-code-with-black-4c87)

As I understand it is not as stringent as [PEP8](https://pep8.org/). For example the default maximum line length is 88 characters.

I tested that it is very easy to install directly as a plugin of an IDE like PyCharm [https://black.readthedocs.io/en/stable/editor_integration.html](https://black.readthedocs.io/en/stable/editor_integration.html) You can even configure that Black reshape the code each time you save a file.

I do not know if Black is the most commonly used tool for code formatting amoung the python community so maybe another choice would be better. But I think a project like PyMoDAQ should make a choice on that matter to prevent (boring ?) discussions about code formatting in the future ;)

**Comment by seb5g** [comment]
I started to use flake8 https://flake8.pycqa.org/en/latest/ 
It is used in the continuous integration of pymodaq :
![image](https://user-images.githubusercontent.com/6609955/109991673-cf89ed80-7d0a-11eb-9792-e137ef188000.png)

**Reply by seb5g** [reply]
everytime a nex commit is pushed, flake8 runs and tells if there are issues. I recently cleaned all pymodaq code to make that linting work. It should also be activate for plugins...

You could also see there is pytest runs. However and this is more crucial the coverage of these tests is rather small

**Comment by quantumm** [comment]
Ok that is nice. I will have a look at flake8 ;)
